File: sceneview/src/main/java/com/google/ar/sceneform/rendering/ViewRenderableInternalData.java
package com.google.ar.sceneform.rendering;

/**
 * Represents shared data used by {@link ViewRenderable}s for rendering. The data will be released
 * when all {@link ViewRenderable}s using this data are finalized.
 */

public class ViewRenderableInternalData {
  private final RenderViewToExternalTexture renderView;

  public ViewRenderableInternalData(RenderViewToExternalTexture renderView) {
    this.renderView = renderView;
  }

  public RenderViewToExternalTexture getRenderView() {
    return renderView;
  }
}

File: sceneview/src/main/java/com/google/ar/sceneform/rendering/ViewRenderableInternalData.kt
package com.google.ar.sceneform.rendering

/**
 * Represents shared data used by [ViewRenderable]s for rendering. The data will be released
 * when all [ViewRenderable]s using this data are finalized.
 */
class ViewRenderableInternalData(private val renderView: RenderViewToExternalTexture) {
    fun getRenderView(): RenderViewToExternalTexture = renderView
}
